tests/Doctrine/Tests/ORM/Query/QueryTest.php

```diff
@@ -251,6 +257,10 @@ public function provideProcessParameterValueIterable(): Generator
         yield 'simple_array' => [$baseArray];
         yield 'doctrine_collection' => [new ArrayCollection($baseArray)];
         yield 'generator' => [$gen()];
+
+        if (PHP_VERSION_ID >= 80100) {
+            yield [array_map([City::class, 'from'], $baseArray)];
```

Comment:
```suggestion
            yield 'enum' => [array_map([City::class, 'from'], $baseArray)];
```

Let's add phpdoc to the generator to enforce this?

Reply:
I've added a `@psalm-return` annotation.

Reply:
I've fixed `ParameterTypeInfererTest` as well.